scripts/src/java/org/oppia/android/scripts/coverage/CoverageReporter.kt

```diff
@@ -31,200 +38,535 @@ class CoverageReporter(
-          border-radius: 2px;
-          display: inline-block;
+      } else if (exemptedFailureTableRows.isNotEmpty()) {
+        append("\n\n")
+        append(tableHeader)
+        append("\n|Exempted :small_red_triangle_down:|\n")
+        append(exemptedFailureTableRows)
+      }
+    }
+
+    val successMarkdownEntries = if (successTableRows.isNotEmpty() ||
+      exemptedSuccessTableRows.isNotEmpty()
+    ) {
+      val detailsContent = buildString {
+        append("\n\n")
+        append("<details>\n")
+        append("<summary>Succeeded Coverages</summary><br>\n\n")
+        if (successTableRows.isNotEmpty()) {
+          append(tableHeader)
+          append(successTableRows)
+          if (exemptedSuccessTableRows.isNotEmpty()) {
+            append("\n|Exempted :small_red_triangle_down:|\n")
+            append(exemptedSuccessTableRows)
+          }
+        } else if (exemptedSuccessTableRows.isNotEmpty()) {
+          append(tableHeader)
+          append("\n|Exempted :small_red_triangle_down:|\n")
+          append(exemptedSuccessTableRows)
         }
-        .legend .covered {
-          background-color: #c8e6c9; /* Light green */
+        append("\n</details>")
+      }
+      detailsContent
+    } else ""
+
+    val testFileExemptedSection = buildString {
+      if (testFileExemptedCasesList.isNotEmpty()) {
+        append("\n\n")
+        append("### Test File Exempted Cases\n")
+        append(testFileExemptedCasesList)
+      }
+    }
+
+    val finalReportText = "## Coverage Report\n\n" +
+      "- Number of files assessed: ${coverageReportContainer.coverageReportList.size}\n" +
+      "- Coverage Analysis: $status" +
+      failureMarkdownTable +
+      failureMarkdownEntries +
+      successMarkdownEntries +
+      testFileExemptedSection
+
+    val finalReportOutputPath = mdReportOutputPath?.let {
+      it
+    } ?: "$repoRoot/coverage_reports/CoverageReport.md"
+
+    File(finalReportOutputPath).apply {
+      parentFile?.mkdirs()
+      writeText(finalReportText)
+    }
+  }
+
+  private fun checkCoverageStatus(): CoverageCheck {
+    coverageReportContainer.coverageReportList.forEach { report ->
+      if (report.hasFailure()) { return CoverageCheck.FAIL }
+
+      if (report.hasDetails()) {
+        val details = report.details
+        val filePath = details.filePath
+        val totalLinesFound = details.linesFound
+        val totalLinesHit = details.linesHit
+
+        val coveragePercentage = calculateCoveragePercentage(
+          totalLinesHit, totalLinesFound
+        )
+
+        val exemption = testFileExemptionList[filePath]
+        if (coveragePercentage < MIN_THRESHOLD) {
+          if (exemption != null) {
+            val ovveriddenMinCoverage = exemption.overrideMinCoveragePercentRequired
+            if (coveragePercentage < ovveriddenMinCoverage) {
+              return CoverageCheck.FAIL
+            }
+          } else {
+            return CoverageCheck.FAIL
+          }
```

Comment:
Can probably be simplified by doing something like:

```kotlin
val threshold = testFileExemptionList[filePath]?.overrideMinCoveragePercentRequired ?: MIN_THRESHOLD
if (coveragePercentage < threshold) return CoverageCheck.FAIL
```

Also: typo on ovveridden.

Reply:
Ah! Thanks. Updated the same.